### ISSUE TITLE
(Cel 6, Sing.Milestones) multiple missing boosts?

### ISSUE THREAD

**Issue by LunavanYeethoven** [antimatter-dimensions/mobile-issues#542]
Some of the singularity milestones seems to be missing or rather don't seem to work. (Some may only be visual bug tho)

- Rewards of reducing DM Ascension interval multiplier seems to be of visual nature, the first four tiers look like they don't change anything, even tho it seems just to round down ~1.2e3 to 1e3. The fifth tier showed 950x suddenly iirc so thats my guess. (Check screenshot 1, it looks like its supposed to show around 1.15e3 instead, and i'm pretty sure the actual multiplier works, as i had >10s after ascensions earlier)
- single-tier Rewards that boosts DM and DE based on certain resources seems to be partly broken. Infinities seems to work fine, TT showed 1x, but needs to be checked again, if it actually still gives a multiplier, Game Speed shows 1x and also gives no boost at all from what i experienced. For latest case (check screenshot 2 (right before) and screenshot 3 (right after) 

![Screenshot_20230301-014701_Antimatter Dimensions](https://user-images.githubusercontent.com/125640626/222024011-bb2dab6e-ac96-4f0d-a278-6de2408c0092.jpg)

![Screenshot_20230301-151652_Antimatter Dimensions](https://user-images.githubusercontent.com/125640626/222170092-fb1f5b46-67b6-448b-b716-014461964317.jpg)

![Screenshot_20230301-151657_Antimatter Dimensions](https://user-images.githubusercontent.com/125640626/222170131-06dd68ab-c1bf-4bb6-987d-f002c8d2cc20.jpg)

**Comment by LunavanYeethoven** [comment]
(Edit Mar/01/2023 3:32pm CET:)
- changed title to reflect singularity milestones as a whole
- added more informations to first issue (-> only visual/rounding issue)
- added a second part to this issue (DE/DM multi based on resources)
- added screenshot 2 and 3

**Comment by kajfik** [comment]
Fixed the first part of this issue in 2.111.0i14.

> single-tier Rewards that boosts DM and DE based on certain resources seems to be partly broken. Infinities seems to work fine, TT showed 1x, but needs to be checked again, if it actually still gives a multiplier, Game Speed shows 1x and also gives no boost at all from what i experienced. For latest case (check screenshot 2 (right before) and screenshot 3 (right after)

To fix this I will need a save to reproduce it.

The formula for DM/DE from TT is:
```
sqrt(max((Currency.timeTheorems.value.log10() - 1000) / 50.0, 1.0))
```
The formula for DM/DE from gamespeed is:
```
max(log10(Game.getGameSpeedupFactor() / 1e120) / 40, 1.0)
```

**Comment by kajfik** [comment]
can't reproduce
